scripts/configurator/parser/cmake_var.py

```diff
@@ -0,0 +1,40 @@
+from typing import Dict, Any
+
+def generate_cmake_var(type :str, cmake_var_str):
+   "factory function that associates the correct type of variable"
+   cmake_type_str_to_class = {
+      "BOOL" : cmake_var_bool,
+      "INTERNAL" : cmake_var_generic,
+      "FILEPATH": cmake_var_generic,
+      "STRING" : cmake_var_generic,
+   }
+   class_obj = cmake_type_str_to_class[type]
+   return class_obj(cmake_var_str)
+
+
+class cmake_var_generic:
```

Comment:
Sorry for the confusion, in case I haven't been clear enough. I meant to use `cmake_var_generic` if you didn't want to have a base class + inheritance, because that's why I saw in the code and I told you it was fine, even if it wouldn't work on other languages.

Now that you made `cmake_var_bool` to inherit from `cmake_var_generic`, the name `cmake_var_generic` is not good enough. It has to be `cmake_var` like you previously named it or `cmake_var_base` at most. 

In conclusion, either you call it `cmake_var_generic` but don't use inheritance, or you call it `cmake_var` and you do use inheritance. The previous change was a mix (`cmake_var` but without inheritance) and was semantically weird.